NDTensors/src/dense/dense.jl

```diff
@@ -68,7 +68,12 @@ end
 
 function Dense(data::DataT) where {DataT<:AbstractArray{<:Any,N}} where {N}
   #println("Warning: Only vector based datatypes are currenlty supported by Dense. The data structure provided will be vectorized.")
-  return Dense(vec(data))
+  ## Vec has an issue where it makes data a `ReshapedArray` this can 
+  ## cause problems in the dispatch system when there are many wrapper types.
+  ## See test/diag.jl line 47
+  v = set_ndims(unwrap_type(data), 1)(undef, length(data))
+  copyto!(expose(v), expose(data))
+  return Dense(v)
```

Comment:
Ah I hadn't considered the no copy aspect. Let me try this, I believe that it will work. Thanks!

Reply:
Both of these codeblocks seem to work fine with the problem case that I ran into in `test/blocksparse.jl` I also did a quick profile and they both use minimal allocations. I even tried changing the block sizes to be significantly larger and they still had a very small number of allocations (about 100 regardless of the size). I am going with the second block because it is more compact

Reply:
It looks like there may be an issue with this code and can be found in the `test_itensor.jl` test. I am working on it now (I also left a comment in the code) I unresolved the conversation to remind

Reply:
Closing this since we have another conversation below